Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs

```diff
@@ -281,6 +289,18 @@ public Task<BoolResult> RequestCopyFileAsync(OperationContext context, ContentHa
             {
                 var location = hashInfo.Locations[replicaIndex];
 
+                //Currently everytime we increment attemptCount's value, we go through every location in hashInfo and try to copy.
+                //We add one because replicaIndex is indexed from zero.
+                //If we reach over maximum retries, return an put result stating so, and no longer retry
+                if ((attemptCount * hashInfo.Locations.Count + replicaIndex + 1) > MaxRetryCount)

```

Comment:
hmm... what if "hashInfo.Locations.Count is 100"? In this case, I think, this logic will prevent us from trying the copy even single time...

Is it the case, or I'm missing something?